### PR TITLE
add more description on running the project in README

### LINKED ISSUE #149: Write a procedure to install the project locally
- create all.yml from all.sample.yml
- edit `remote_user` and `project_path`
- create `tmp` directory in ansible directory`
- `make setup-ansible`
- `make up`
- `make migrate`

## create a super
- `make shell`
- `./manage createsuperuser`

### PR DESCRIPTION
**Pull request by boney-bun**
fix #149

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+## Miscellaneous
+
+### Populate Initial Data
+#### Add superuser
+Admin user is required to administer the site.
+To add admin:
+
+```
+# go to deployment folder
+make shell
```
Comment (pierrealixt) [inline README.md]: we can run `make superuser`

File: README.md [hunk 2]
```diff
@@ -61,6 +67,76 @@ make migrate
 make collectstatic
 ```
 
+### 2. Run the server
+#### A. From PyCharm Professional
+```
+# go to deployment/ansible/development/group_vars
```
Comment (pierrealixt) [inline README.md]: remove this line but keep the `cd` command

File: README.md [hunk 3]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+cp all.sample.yml all.yml
+```
+
+- edit line 6 (*remote_user*), 8 (*remote_group*), and 10 (*project_path*) in all.yml accordingly
+  - make sure that *remote_user* align with your local user
+  - *remote_group* is likely not change if using linux and macOS
+  - *project_path* is equal to with your cloned directory
+
+```
+# go to deployment/ansible
```
Comment (pierrealixt) [inline README.md]: remove this line. add a `cd` command instead

File: README.md [hunk 4]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+- Open PyCharm
+- Notice your pycharm, there should be *Ford3* django server in the toolbar.
+  - Wait for a couple of minutes. Make sure the PyCharm has loaded all the necessary files.
+  - If pycharm requires to install additional supported modules, click on the provided link
+- Click on the `play button` or `debug button` next to *Ford3* instance in the toolbar. The pycharm will run the server for you
+  - sometimes restart pycharm can remedy the problem too
+  - If the server doesn't run, try:
+
+```
+# from deployment folder
```
Comment (pierrealixt) [inline README.md]: remove this line. `cd deployment`

File: README.md [hunk 5]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+make down
+make up
+```
+
+
+#### B. From CLI
+```
+# go to deployment folder
```
Comment (pierrealixt) [inline README.md]: remove. `cd deployment`

File: README.md [hunk 6]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+
+## Miscellaneous
+
+### Populate Initial Data
+#### Add superuser
+Admin user is required to administer the site.
+To add admin:
+
+```
+# go to deployment folder
```
Comment (pierrealixt) [inline README.md]: remove. `cd deployment`

File: README.md [hunk 7]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+make shell
+python manage.py createsuperuser
+# write your usename, email, and password
+```
+#### Add initial data
+```
+# go to deployment folder
```
Comment (pierrealixt) [inline README.md]: remove. `cd deployment`

File: README.md [hunk 8]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+Admin user is required to administer the site.
+To add admin:
+
+```
+# go to deployment folder
+make shell
+python manage.py createsuperuser
+# write your usename, email, and password
+```
+#### Add initial data
```
Comment (pierrealixt) [inline README.md]: that should go before running the server

File: README.md [hunk 9]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+#### 3. Open Browser
+
+- Open browser and type: `http://localhost`
+- You have the project running now
+
+
+## Miscellaneous
+
+### Populate Initial Data
+#### Add superuser
```
Comment (pierrealixt) [inline README.md]: That should go before running the server

File: README.md [hunk 10]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+
+- edit line 6 (*remote_user*), 8 (*remote_group*), and 10 (*project_path*) in all.yml accordingly
+  - make sure that *remote_user* align with your local user
+  - *remote_group* is likely not change if using linux and macOS
+  - *project_path* is equal to with your cloned directory
+
+```
+# go to deployment/ansible
+mkdir tmp
+make setup-ansible
```
Comment (pierrealixt) [inline README.md:105]: it fails here.
after `mkdir` we are still in `ford3/deployment/ansible`
`cd ..`
`make setup-ansible`

File: README.md [hunk 11]
```diff
@@ -61,6 +67,76 @@ make migrate
...
+# from deployment folder
+make down
+make up
+```
+
+
+#### B. From CLI
+```
+# go to deployment folder
+make migrate
```
Comment (pierrealixt) [inline README.md:127]: I also had the warning so maybe we don't need to run them at the beginning of the setup

**Review by pierrealixt — COMMENTED** [review]
We should add a section about running the tests:

In order to run the selenium tests, make sure the web server is not running.
`cd deployment`
`make selenium-up`
`make test`

**Comment by pierrealixt** [comment]
We should remove as well the waffle widget. Waffle is no more alive.

It shows the Inasafe build status not ford3.
Screen Shot 2019-05-17 at 11 28 14

We can remove the section about google authentication.

**Comment by boney-bun** [comment]
@pierrealixt do we need to activate landscape account? 
if so, perhaps you can activate the account in: https://landscape.io/github/kartoza/ford3
otherwise, i'll remove it.

File: README.md [hunk 12]
```diff
@@ -25,32 +25,37 @@ The latest source code is available at
...
+Therefore, the project will be in the `/home/web/ford3` directory.
+
+The project directory `/home/web/ford3` will be called and referred as `<parent>` throughout the rest of doc below.
+
+### 1. Build the images and set up the docker for the project
 ```
+cd /home/web
 git clone git://github.com/kartoza/ford3.git
-cd ford3/deployment
+cd *<parent>*/deployment
```
Comment (pierrealixt) [inline README.md]: why parent?
The name of the repo is `ford3` so `cd ford3/deployment` is correct

Comment (boney-bun) [reply README.md]: it aims to shorten `ford3/deployment`.
i refer to this folder  in many lines.

do you want to use the long version instead?

Comment (pierrealixt) [reply README.md]: assume that the user is already in `ford3`

File: README.md [hunk 13]
```diff
@@ -25,32 +25,35 @@ The latest source code is available at
...
+3. Run the server
+4. Open browser
+
+Suppose the current directory is `/home/web`.
+
+Therefore, the project will be in the `/home/web/ford3` directory.
+
+### 1. Build the images and set up the docker for the project
 ```
+cd /home/web
```
Comment (pierrealixt) [inline README.md]: you're assuming the user user having this directory on his computer?

**Review by pierrealixt — APPROVED** [review]
LGTM

**Pull request merged**